components/payments/cmd/api/internal/api/module.go

```diff
@@ -0,0 +1,141 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"net/http"
+	"runtime/debug"
+	"strconv"
+
+	"github.com/formancehq/payments/cmd/api/internal/storage"
+	"github.com/formancehq/stack/libs/go-libs/api"
+	"github.com/formancehq/stack/libs/go-libs/httpserver"
+	"github.com/formancehq/stack/libs/go-libs/logging"
+	"github.com/formancehq/stack/libs/go-libs/otlp"
+	"github.com/gorilla/mux"
+	"github.com/rs/cors"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+	"go.uber.org/fx"
+)
+
+const (
+	otelTracesFlag = "otel-traces"
+	serviceName    = "Payments"
+)
+
+func HTTPModule(serviceInfo api.ServiceInfo, bind string) fx.Option {
+	return fx.Options(
+		fx.Invoke(func(m *mux.Router, lc fx.Lifecycle) {
+			lc.Append(httpserver.NewHook(bind, m))
+		}),
+		fx.Supply(serviceInfo),
+		fx.Provide(httpRouter),
+	)
+}
+
+func httpRecoveryFunc(ctx context.Context, e interface{}) {
+	if viper.GetBool(otelTracesFlag) {
+		otlp.RecordAsError(ctx, e)
+	} else {
+		logrus.Errorln(e)
+		debug.PrintStack()
+	}
+}
+
+func httpCorsHandler() func(http.Handler) http.Handler {
+	return cors.New(cors.Options{
+		AllowedOrigins:   []string{"*"},
+		AllowedMethods:   []string{http.MethodGet, http.MethodPost, http.MethodPut},
+		AllowCredentials: true,
+	}).Handler
+}
+
+func httpServeFunc(handler http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+		handler.ServeHTTP(w, r)
+	})
+}
+
+func handleStorageErrors(w http.ResponseWriter, r *http.Request, err error) {
+	if errors.Is(err, storage.ErrDuplicateKeyValue) {
+		handleErrorBadRequest(w, r, errors.New("unique reference already existing"))
+	} else if errors.Is(err, storage.ErrNotFound) {
+		handleNotFoundError(w, r, err)
+	} else {
+		handleServerError(w, r, err)
+	}
+}
+
+func handleErrorBadRequest(w http.ResponseWriter, r *http.Request, err error) {
```

Comment:
Maybe use shared lib for all these response?

Reply:
Done thanks !